src/Command/DataFlow/DataFlowProduceJobsCommand.php

```diff
@@ -96,25 +100,17 @@ protected function execute(InputInterface $input, OutputInterface $output)
      */
     private function getDataFlowsToRun(array $dataFlowCandidates): array
     {
-        $now = new \DateTime();
-
-        return array_filter($dataFlowCandidates, function (DataFlow $dataFlow) use ($now) {
-            // If data flow hasn't run yet at all is should do it now
-            if (empty($dataFlow->getLastRunAt())) {
-                return true;
-            }
+        return array_filter($dataFlowCandidates, function (DataFlow $dataFlow) {
 
             // If there already is an active job (not completed or failed jobs), if so we should not schedule a new job
             $activeJobs = $this->dataFlowJobRepository->findActiveJobsByDataFlow($dataFlow);
             if (!empty($activeJobs)) {
                 return false;
             }
 
-            // Difference in seconds between now and last time the data flow ran
-            $seconds = $now->getTimestamp() - $dataFlow->getLastRunAt()->getTimestamp();
-
-            if ($dataFlow->getFrequency() < $seconds) {
-                return true;
+            if (null !== $dataFlow->getSchedule()) {
+                $cron = CronExpression::factory(\strval($dataFlow->getSchedule()));
```

Comment:
Is `strval` needed here? What is the type of `$dataFlow->getSchedule()` (if not null)?

Reply:
It actually a CronExpression object, so I don't need to use strval, but my editor didn't recognize it at first.